src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs

```diff
@@ -246,6 +246,23 @@ private static void VerifyBytes(IntPtr p, int length)
         }
 
         [Fact]
+        public static void ValidateExceptionForHRAPI()
+        {
+            int errorCode = -2146231029;
+            COMException getHRException = Marshal.GetExceptionForHR(errorCode) as COMException;
+            Assert.NotNull(getHRException);
+            try
+            {
+                Marshal.ThrowExceptionForHR(errorCode);
+            }
+            catch (COMException e)
+            {             
+                Assert.Equal(e.HResult, errorCode);
+                Assert.Equal(e.HResult, getHRException.HResult);
```

Comment:
Isn't a different think , in the catch block i am comparing the HResults from both exceptions.I intention is to compare the hresult from e and getHRException not compare it to errorCode